src/main/kotlin/com/playwright/remote/engine/browser/RemoteBrowser.kt

```diff
@@ -1,34 +1,30 @@
 package com.playwright.remote.engine.browser
 
-import com.google.gson.JsonObject
 import com.playwright.remote.engine.browser.api.IBrowser
 import com.playwright.remote.engine.browser.impl.Browser
-import com.playwright.remote.engine.processor.ChannelOwner
 import com.playwright.remote.engine.processor.MessageProcessor
 import com.playwright.remote.engine.websocket.WebSocketTransport
+import okio.IOException
 
-class RemoteBrowser(parent: ChannelOwner, type: String, guid: String, initializer: JsonObject) :
-    ChannelOwner(parent, type, guid, initializer) {
-
-    private fun browser(): IBrowser =
-        messageProcessor.getExistingObject(initializer["browser"].asJsonObject["guid"].asString)
+class RemoteBrowser {
 
     companion object {
         @JvmStatic
         fun connectWs(wsEndpoint: String): IBrowser {
             val webSocketTransport = WebSocketTransport(wsEndpoint)
             val messageProcessor = MessageProcessor(webSocketTransport)
-            val remoteBrowser = messageProcessor.waitForObjectByGuid("remoteBrowser") as RemoteBrowser
-            val browser = remoteBrowser.browser() as Browser
-
-            val connectionCloseListener: (WebSocketTransport) -> Unit = { browser.notifyRemoteClosed() }
+            val browser = messageProcessor.waitForLaunchedBrowser()
+            browser as IBrowser
```

Comment:
Just inline this instruction with the line above.
Line above may be line this 
`val browser = messageProcessor.waitForLaunchedBrowser() as IBrowser`

Reply:
Fixed